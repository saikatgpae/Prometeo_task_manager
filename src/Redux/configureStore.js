import { combineReducers } from 'redux';
import { configureStore } from '@reduxjs/toolkit';
import tasksReducer from './Tasks/tasksRedux';

const rootReducer = combineReducers({
  tasks: tasksReducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
