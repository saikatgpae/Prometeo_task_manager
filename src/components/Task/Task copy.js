/* eslint-disable */
import React, { useRef, useState } from 'react';
import uuid from 'react-uuid';
import './Task.css';

export default function Task(prop) {
//   const [editor, setEditor] = useState(false);
  const [updatedInput, setUpdatedInput] = useState();
  const [taskID, setTaskID] = useState();
  const checkBoxRef = useRef(null);
  const taskNameRef = useRef(null);
  const {
    tasks,
  } = prop;

  const viewMode = {};
  const editMode = {};

  //   Edit button clicked
  const handleEditClick = (e) => {
    e.preventDefault();
    e.target.classList.add('none');
    e.target.previousSibling.previousSibling.classList.add('none');
    e.target.nextSibling.classList.remove('none');
    e.target.previousSibling.classList.remove('none');
  };
  //   Save button clicked
  const handleSaveClick = (e) => {
    e.preventDefault();
    e.target.classList.add('none');
    e.target.previousSibling.classList.remove('none');
    e.target.previousSibling.previousSibling.classList.add('none');
    e.target.previousSibling.previousSibling.previousSibling.classList.remove('none');
    console.log(e.target.id.split('-')[1]);
    console.log(taskNameRef.current.value);
  };

  //   Handle onchange of input form
  const handleChange = (e) => {
    console.log(e.target.value);
    // setUpdatedInput(e.target.value);
    // console.log(updatedInput);
  };

  return (
    <div>
      {
        (tasks)
          ? tasks.map((task, index) => (
            <ul style={{ listStyleType: 'none' }} className="d-flex justify-content-around" key={uuid()}>
              <label htmlFor={index}>
                complete?
                <input ref={checkBoxRef} style={{ accentColor: 'green' }} type="checkbox" defaultChecked={task.complete} id={index} />
              </label>
              <li style={viewMode}>{task.taskName}</li>
              <input id={`edit-${index}`} ref={taskNameRef} onChange={handleChange} className="none" style={editMode} type="text" defaultValue={task.taskName} />
              <button id={`edit-${index}`} style={viewMode} onClick={handleEditClick} type="button" className="btn btn-primary">Edit</button>
              <button id={`save-${index}`} style={editMode} onClick={handleSaveClick} type="button" className="none btn btn-primary">save</button>
              <button type="button" className="btn btn-danger">Delete</button>
            </ul>
          )) : <p>Do not hav a task? Add one.</p>
      }

    </div>
  );
}


