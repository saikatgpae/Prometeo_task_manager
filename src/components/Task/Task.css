.none {
  display: none !important;
}

.strike {
  text-decoration: line-through;
}

.delete-button {
  color: red;
}

.save-button,
.edit-button,
.delete-button {
  border: none;
  float: inline-end;
}

.save-button:hover,
.edit-button:hover,
.delete-button:hover {
  text-decoration: underline;
}

.edit-button:disabled {
  text-decoration: none;
}

/* dive tag holding the pag */
.container-div {
  color: black;
}

/* select tag filter */

.filter-menu {
  background-color: whitesmoke;
  float: inline-start;
}

/* div tag holding the table */
.table-div {
  border-radius: 5px;
  height: 400px;
  width: 100%;
  background-color: white;
}

/* table tag */
.task-table {
  /* display: flex; */

  /* width: 100%; */
}

/* tr tag row data */
.table-data {
  /* color: black; */
}

/* td tag */
.task-display {
  /* padding: 0 !important; */
}

/* input tag for update */
.task-input {
  border: none;
  background-color: rgb(253, 246, 103);
}

/* strong tag to display task */
.task-name {
  /* width:335px; */

  /* overflow:hidden; */

  /* width: 100px; */
}

@media screen and (max-width: 767px) {
  .container {
    padding: 0 !important;
  }
}
