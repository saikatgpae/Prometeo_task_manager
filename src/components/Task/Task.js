// /* eslint-disable */
import React from 'react';
import uuid from 'react-uuid';
import './Task.css';
import { useDispatch } from 'react-redux';
import {
  completeTask, deleteTask, editTask, incompleteTask,
} from '../../Redux/Tasks/tasksRedux';

export default function Task(prop) {
  const dispatch = useDispatch();
  const { tasks } = prop;

  //   Edit button clicked
  const handleEditClick = (e) => {
    const elementId = e.target.id.split('-')[1];
    e.target.classList.add('none');
    document.getElementById(`save-${elementId}`).classList.remove('none');
    document.getElementById(`update-${elementId}`).classList.remove('none');
    document.getElementById(`task-${elementId}`).classList.add('none');
  };
  //   Save button clicked
  const handleSaveClick = (e) => {
    const elementId = e.target.id.split('-')[1];
    const task = document.getElementById(`update-${elementId}`).value;
    dispatch(editTask(task, elementId));
  };

  //   Handle Delete butoon click
  const handleDelete = (e) => {
    const indexNumber = Number(e.target.id.split('-')[1]);
    dispatch(deleteTask(indexNumber));
  };

  //   Handle the check box change
  const handleChang = (e) => {
    const checkedValue = e.target.checked;
    const arrIndex = Number(e.target.id.split('-')[1]);
    // eslint-disable-next-line
    (checkedValue) ? dispatch(completeTask(arrIndex)) : dispatch(incompleteTask(arrIndex));
  };

  return (
    <div>
      {
        (tasks)
          ? tasks.map((task, index) => (
            <ul style={{ listStyleType: 'none' }} className="d-flex justify-content-around" key={uuid()}>
              <label htmlFor={index}>
                complete?
                <input onChange={handleChang} style={{ accentColor: 'green' }} type="checkbox" defaultChecked={task.complete} id={`status-${index}`} />
              </label>
              {
                (!task.complete) ? <li id={`task-${index}`}>{task.taskName}</li> : <li id={`task-${index}`}><s>{task.taskName}</s></li>
              }
              <input id={`update-${index}`} className="none" type="text" defaultValue={task.taskName} />
              <button disabled={task.complete} id={`edit-${index}`} onClick={handleEditClick} type="button" className="btn btn-primary">Edit</button>
              <button id={`save-${index}`} onClick={handleSaveClick} type="button" className="none btn btn-primary">Save</button>
              <button onClick={handleDelete} id={`delete-${index}`} type="button" className="btn btn-danger">Delete</button>
            </ul>
          )) : <p>Do not hav a task? Add one.</p>
      }

    </div>
  );
}
