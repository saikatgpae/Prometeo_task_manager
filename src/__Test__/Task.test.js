// import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import Task from '../components/Task/Task';

const initialState = [
  { complete: false, taskName: 'playing' },
  { complete: false, taskName: 'drawing' },
  { complete: false, taskName: 'Shopping' },
];

const mockStore = configureStore([]);
describe('My Connected React-Redux Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore(initialState);

    component = renderer.create(
      <Provider store={store}>
        <Task />
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should dispatch an action on button click', () => {

  });
});
