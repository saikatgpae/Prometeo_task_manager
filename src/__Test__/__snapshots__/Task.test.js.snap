// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks render correctly 1`] = `
<div
  className="container container-div"
>
  <select
    className="filter-menu"
    data-testid="select-menu-input"
    id="filter-menu"
    onChange={[Function]}
  >
    <option>
      all
    </option>
    <option>
      complete
    </option>
    <option>
      incomplete
    </option>
  </select>
  <div
    className="table-responsive table-div"
  >
    <table
      className="task-table table table-hover"
      id="taskTable"
    >
      <tbody>
        <tr
          className="table-data"
        >
          <td
            className=""
          >
            <input
              className="checkBox"
              defaultChecked={false}
              id="status-0"
              onChange={[Function]}
              style={
                Object {
                  "accentColor": "green",
                }
              }
              type="checkbox"
            />
          </td>
          <td
            className="task-display"
          >
            <strong
              className="task-name text-primary"
              id="task-0"
            >
              playing
            </strong>
            <input
              className="none task-input"
              defaultValue="playing"
              id="update-0"
              required={true}
              type="text"
            />
          </td>
          <td
            className=""
          >
            <button
              className="edit-button"
              disabled={false}
              id="edit-0"
              onClick={[Function]}
              type="button"
            >
              Edit
            </button>
            <button
              className="none save-button"
              id="save-0"
              onClick={[Function]}
              type="button"
            >
              Save
            </button>
          </td>
          <td
            className=""
          >
            <button
              className="delete-button"
              id="delete-0"
              onClick={[Function]}
              type="button"
            >
              Delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
